app/services/catalog_record_id.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+class CatalogRecordId
+  # NOTE: This block is a junk drawer of simple class methods meant to help
+  #       switch between Symphony and Folio catalog implementations, and can be
+  #       dumped or in-lined once we're done with Symphony for good.
+  class << self
+    def other_links(model)
+      new(model).other_links.map(&:catalogRecordId)
+    end
+
+    def valid?(catalog_record_ids)
+      catalog_record_ids.all? { |id| id.match?(Regexp.new(pattern_string)) }
+    end
+
+    def label
+      return "Folio ID" if Settings.enabled_features.folio
+
+      "Catkey"
+    end
+
+    def manage_label
+      return "Manage Folio ID" if Settings.enabled_features.folio
+
+      "Manage catkey"
+    end
+
+    def index_field
+      return SolrDocument::FIELD_FOLIO_INSTANCE_HRID if Settings.enabled_features.folio
+
+      SolrDocument::FIELD_CATKEY_ID
+    end
+
+    def pattern_string
+      return "\\A(L|a|in)[0-9]+\\z" if Settings.enabled_features.folio
+
+      "\\A[0-9]+(:[0-9]+)*\\z"
+    end
+
+    def html_pattern_string
+      return "^(L|a|in)[0-9]+$" if Settings.enabled_features.folio
+
+      "^[0-9]+(:[0-9]+)*$"
+    end
+
+    def index_field_prefix
+      return "folio" if Settings.enabled_features.folio
+
+      "catkey"
+    end
+
+    def value
+      return "folio" if Settings.enabled_features.folio
+
+      "symphony"
+    end
+
+    def previous_value
+      return "previous folio" if Settings.enabled_features.folio
+
+      "previous symphony"
+    end
+  end
```

Comment:
why the `class << self` section for some class methods, and the `self.` definition for others?

Reply:
Good question! I thought it might be useful to put all the stuff related to switching between implementations into a single syntactic block for easier removal/reconsideration later.

Reply:
Already is, at the top of the block.